pkg/storage/kubernetes/ipam.go

```diff
@@ -417,11 +418,11 @@ func IPManagementKubernetesUpdate(mode int, ipam *KubernetesIPAM, ipamConf where
 	var pool storage.IPPool
 	var err error
 
-	ctx, cancel := context.WithTimeout(context.Background(), storage.RequestTimeout)
-	defer cancel()
+	requestCtx, requestCancel := context.WithTimeout(ctx, storage.RequestTimeout)
```

Comment:
What about renaming these variables ? `ctxWithTimeout` or something like that.

Reply:
Since timeout check is still done on ctx instead of requestCtx, is it really required to have a new requestContext.

Reply:
> What about renaming these variables ? ctxWithTimeout or something like that.

I rather my naming tbh. Can I ask why you want to add the Timeout? Thank Miguel.

> Since timeout check is still done on ctx instead of requestCtx, is it really required to have a new requestContext.

The only good I can see that it does is make sure there isn't any dangling / hanging connections to the backend. @maiqueb  What do you think?